src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -188,8 +188,10 @@ private void BatchTrafficWatch(ArraySegment<BatchRequestParser.CommandData> pars
 
         private async Task HandleClusterTransaction(DocumentsOperationContext context, MergedBatchCommand command, ClusterTransactionCommand.ClusterTransactionOptions options)
         {
+            var guid = GetRaftGuidFromHeaders();
```

Comment:
This is a great example of why we don't want it in the headers. 
It is very easy to forget to do that in various places, and lose the protection it gives us

Reply:
But even when passed via the url it has to be retrieved.. so it is the same, no?

Reply:
Yes, so don't put it in the URL. Make it in the command body itself.